Enable syntax-highlighted diffs

mmirus opened the pull request:
Hi there! Thanks for the theme!

This PR allows syntax highlighting to be used in diffs, and sets some default diff background colors that go well with that change. The result is (intentionally) similar to GitHub's diff rendering.

Hope this is helpful!

Before:

![image](https://user-images.githubusercontent.com/524994/140584364-5e10efd8-0a68-4115-81cd-d510a5a05cfb.png)

After:

![image](https://user-images.githubusercontent.com/524994/140584865-76456be2-1977-48f0-a859-c20fe2aa295d.png)

Instead of changing the theme's defaults, I imagine it would also be possible to allow the fg colors I removed to be disabled from the plugin's settings, so that users would have the option to use syntax highlighting if they wanted.

## Changes

- Remove the fg color on diff lines so that normal syntax colors are used
- Use darker background colors for the diffs, so that the text (which is no longer dark) is still legible

## Notes

- If there is already a way to override the fg colors set for diffs [here](https://github.com/olimorris/onedarkpro.nvim/blob/main/lua/onedarkpro/theme.lua#L34-L37) without modifying the theme files, then this PR isn't needed. I couldn't figure out how.
- I just implemented this and it's barely tested. I'm sure there are ways to improve the colors I chose!
- I haven't added anything equivalent to the light theme.

olimorris commented:
Firstly, thanks for the PR. Really appreciated.

Secondly, very happy to just merge this into the main repo. You could override my defaults using `hlgroup` overrides. I would normally advocate that as colors are very opinionated and I've tried to stay true to `One Dark Pro's`. However the defaults for the `diff` highlight groups, suck 😆. In its current form, it would break for the light color scheme so I'll add the corresponding ones.

mmirus commented:
Awesome! I'm glad it's helpful.

I'm happy to pop open the light version and add some colors later today if you haven't by then.

> You could override my defaults using hlgroup overrides.

Overriding the backgrounds would be possible this way, but it's it possible to "unset" the foreground colors specified in the Diff* rules so that they other rules apply? I could only figure out how to set them to a new color (which still doesn't give correct syntax highlighting), not remove those values entirely.

mmirus commented:
Made a couple changes:

1. Updated the dark diff colors to imitate One Dark Pro in vscode instead of GitHub's diffs:
    ![image](https://user-images.githubusercontent.com/524994/140615067-18a78321-d1de-4596-8dd3-cea9ab845a05.png)
2. Added diff colors for the light theme
    ![image](https://user-images.githubusercontent.com/524994/140615716-5be77687-addd-4552-9012-23370a73be99.png)

olimorris commented:
Once again thanks so much. Output looks awesome. I've added your name to the readme 👍🏼

mmirus commented:
Awesome! Thanks for being so welcoming.
